10up/autoshare-for-twitter
# Improvements in tweet message character count.

### Description of the Change
PR contains the below changes. 

- [x] Short warning when message getting long
- [x] Error when message is too long
- [x] Set max character for the text area, so that people can't enter longer Tweets

| Classic Editor | Guternberg |
| --- | --- |
|![Screenshot 2023-01-04 at 11 38 27 AM](https://user-images.githubusercontent.com/10613171/210495597-86a2b0b3-903f-4f8d-a576-f78c7f696ab5.png)|![Screenshot 2023-01-04 at 11 44 06 AM](https://user-images.githubusercontent.com/10613171/210495694-1f84851a-6d8e-4579-bbe0-8e0ed21f3350.png)|
|![Screenshot 2023-01-04 at 11 41 29 AM](https://user-images.githubusercontent.com/10613171/210495601-3fc517d8-8421-4592-818a-219c68358d01.png)|![Screenshot 2023-01-04 at 11 42 27 AM](https://user-images.githubusercontent.com/10613171/210495692-cb04db71-07fa-409b-a39a-f95b8c277b66.png)|
|![Screenshot 2023-01-04 at 11 47 59 AM](https://user-images.githubusercontent.com/10613171/210495708-eaf8e36f-bf25-4d35-bdf8-83c8bf633f88.png)|![Screenshot 2023-01-04 at 11 48 49 AM](https://user-images.githubusercontent.com/10613171/210495719-e543ddc0-4381-4d8f-aa5d-c7892da810f0.png)|

Closes #178 

### How to test the Change
Need to test with Gutenberg and Classic Editor.
1. Click on the new Post to create a post
2. Edit "custom message" for tweet
3. See the `Getting Long!` message after character count 240
4. See the `Too Long!` message once the character count reaches 280
5. Verify you are not able to enter text anymore after the character count reaches 280
6. Verify steps 3-5 with tweet now functionality. 

### Changelog Entry
> Changed - Improvements in tweet message character count.



### Credits

Props @iamdharmesh @Sidsector9 @jeffpaul 


### Checklist:
